### DISCUSSION TITLE
FF-251 - Delete FileSystemItem

### DISCUSSION THREAD

**Discussion by open-schnick** [FileFighter/RestApi#61] [Ideas]
https://dhbw-karlsruhe.myjetbrains.com/youtrack/issue/FF-251

We have a problem for this feature:

What are we gonna do if the user deletes a folder with contents? 

1. are we deleting recursivly? -> performance.
2. what if the user is not allowed to delete files in the folder
3. Worst case there are files that cannot be accessed, anymore because they are not explicitly linked to a folder.

**Comment by qvalentin** [comment]
1. Yes. Everything else would not be user-friendly
2. A) Only the files the user can delete are deleted (meaning the folder will still exist)
    B) Nothing is deleted

Keep in mind that recursive actions will also be needed for permissions editing.

**Reply by open-schnick** [reply]
so we have 3 actual cases:
the user wants to delete:
1. a file -> check for permissions -> delete
2. a folder which contains only editable files and subfolders -> check for permissions -> delete folder and all files.
3. a folder which contains files that are not editable for the user. we could
  - delete all files and the folder (not good, unlinked files.)
  - delete all files and the folder but remap the sub files and folders to a new one which cannot be seen by the user.
  - other ?...

> Only the files the user can delete are deleted (meaning the folder will still exist)

how do we explain the user, that the folder the he wanted to be removed is still there but the contents are deleted? -> he basicly has a empty folder that cannot be deleted.

**Reply by qvalentin** [reply]
> how do we explain the user, that the folder the he wanted to be removed is still there but the contents are deleted? -> he basicly has a empty folder that cannot be deleted.

This would be the case when the user only has the right to seem some files, but not to edit them.

If all the files he can see are gone he should also loose the permission to see the folder (for other users the folder would then still exist).

So one more case (4 in total).

**Reply by open-schnick** [reply]
Ok that is implementable. Can you or @Gimleux implement these cases in a feature file?
